### ISSUE TITLE
some questions

### ISSUE THREAD

**Issue by ivanevgenyev** [AlexeyAB/darknet#1464]
Hello, Alexey!
I worked a little with your project and I had a few questions.

1. When i train YOLOv3 on my custom object i see a string in terminal, like this: 
"9002: 0.211667, 0.060730 avg, 0.001000 rate, 3.868000 seconds, 576128 images Loaded: 0.000000 seconds".
In readme you say "0.060730 avg - average loss (error) - the lower, the better"
What do correspond this value for avg loss (test and/or train)?
Which means the second number (0.211667)?
 
2. Do we need a test files (.jpg and .txt) in the data folder (or just for a train) for training data? 

3. Is it possible to save not every 100th weight, but weights with a minimum avg loss?

4. Is it possible to find out -- what avg loss corresponds to an already saved weights file?

Thank you!
Ivan (Moscow)

**Comment by AlexeyAB** [comment]
@ivanevgenyev Hi,

1. avg loss - is for training.  (0.211667) - is current loss. (0.060730 avg) - is average loss.

2. No.

3. No. Also avg loss is not accurate indicator. The lowest avg loss doesn't mean that mAP will be the highest. 

4. Change this line: https://github.com/AlexeyAB/darknet/blob/8436251a05ec0cfa906089ee022db17498e6f65f/src/detector.c#L212
to this `sprintf(buff, "%s/%s_%d_%f.weights", backup_directory, base, i, avg_loss); `
and recompile, so the last value in the filename will be (avg loss).

(Or you can just start training from this "saved weights file", wait for 20 iterations and see avg loss.)

**Comment by ivanevgenyev** [comment]
Thank you for fast answers, Alexey!
I have another one.
Are the dropout layers used in the architecture of YOLO v3?
Where i can find a description of the network architecture?

**Comment by AlexeyAB** [comment]
@ivanevgenyev 

* Drouput - is a very outdated and not optimal approach. Yolo v1 (2015) used dropout. Yolo v2 and v3 uses Batch normalization: https://github.com/AlexeyAB/darknet/issues/476#issuecomment-374172707

Page 2: https://arxiv.org/pdf/1612.08242v1.pdf

> Batch Normalization. Batch normalization leads to significant improvements in convergence while eliminating the need for other forms of regularization [7]. By adding batch normalization on all of the convolutional layers in YOLO we get more than 2% improvement in mAP. **Batch normalization also helps regularize the model. With batch normalization we can remove dropout from the model without overfitting**

----

* Yolo v3 architecture:
    * https://pjreddie.com/media/files/papers/YOLOv3.pdf
    * The simplest explanation: https://github.com/AlexeyAB/darknet/issues/666#issuecomment-388565246
    * How does Convolutional Neural Network works: https://github.com/AlexeyAB/darknet/issues/796#issuecomment-388553709
